lib/engine/game/g_18_cz.rb

```diff
@@ -164,6 +164,26 @@ def block_lay_for_purple_tiles
           tile.blocks_lay = true if tile.name.end_with?('p')
         end
       end
+
+      def must_buy_train?(entity)
+        !entity.rusted_self &&
+        !depot.depot_trains.empty? &&
+        (entity.trains.empty? ||
+          (entity.type == :medium && entity.trains.none? { |item| !!(item.name =~ medium_train_regex) }) ||
+          (entity.type == :large && entity.trains.none? { |item| !!(item.name =~ large_train_regex) }))
+      end
+
+      def small_train_regex
```

Comment:
let's not do regexes and instead have a dictionary

Reply:
yeah, thats maybe better ;)